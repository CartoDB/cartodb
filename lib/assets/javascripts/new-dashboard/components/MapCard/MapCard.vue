<template>
  <SimpleMapCard
    v-if="!condensed"
    :map="map"
    :isSelected="isSelected"
    :canHover="canHover"
    :selectMode="selectMode"
    @toggleSelection="toggleSelection"/>
  <CondensedMapCard
    v-else
    :map="map"
    :isSelected="isSelected"
    :canHover="canHover"
    :selectMode="selectMode"
    @toggleSelection="toggleSelection"/>
</template>

<script>
import CondensedMapCard from './CondensedMapCard';
import SimpleMapCard from './SimpleMapCard';
import props from './shared/props';

export default {
  name: 'MapCard',
  props: {
    ...props,
    condensed: {
      type: Boolean,
      default: false
    }
  },
  components: {
    SimpleMapCard,
    CondensedMapCard
  },
  methods: {
    toggleSelection () {
      this.$emit('toggleSelection', {
        map: this.$props.map,
        isSelected: !this.$props.isSelected
      });
    }
  }
};
</script>
