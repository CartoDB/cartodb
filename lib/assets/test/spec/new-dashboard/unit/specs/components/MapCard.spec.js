import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import visualization from '../fixtures/visualizations';
import usersArray from '../fixtures/users';
import MapCard from 'new-dashboard/components/MapCard';
import MapQuickActions from 'new-dashboard/components/QuickActions/MapQuickActions';

// Backbone Models
import ConfigModel from 'dashboard/data/config-model';
import UserModel from 'dashboard/data/user-model';

const localVue = createLocalVue();
localVue.use(Vuex);

let $cartoModels, actions, store;
const $t = key => key;

function configCartoModels (attributes = {}) {
  const user = new UserModel(attributes.user || usersArray[0]);
  const config = new ConfigModel(attributes.config || { maps_api_template: 'http://{user}.example.com' });
  return { user, config };
}

describe('MapCard.vue', () => {
  beforeEach(() => {
    $cartoModels = configCartoModels();

    actions = {
      'maps/deleteLike': jest.fn(),
      'maps/like': jest.fn()
    };

    store = new Vuex.Store({
      actions
    });
  });

  it('should render correct contents', () => {
    const map = visualization.visualizations[0];
    const mapCard = shallowMount(MapCard, {
      propsData: { map, isSelected: true },
      mocks: {
        $cartoModels,
        $t
      }
    });

    expect(mapCard).toMatchSnapshot();
  });

  it('should show the map is shared', () => {
    $cartoModels = configCartoModels({ user: usersArray[1] });
    const map = visualization.visualizations[0];
    const mapCard = shallowMount(MapCard, {
      propsData: { map },
      mocks: {
        $cartoModels,
        $t
      }
    });

    expect(mapCard).toMatchSnapshot();
  });

  it('should render description and tags', () => {
    const map = {...visualization.visualizations[0]};
    map.description = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
    map.tags = ['Hi', 'Hello', 'CARTO Rules'];
    const mapCard = shallowMount(MapCard, {
      propsData: { map },
      mocks: {
        $cartoModels,
        $t
      }
    });

    expect(mapCard).toMatchSnapshot();
  });

  it('should render description and tag dropdown', () => {
    const map = {...visualization.visualizations[0]};
    map.description = 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.';
    map.tags = ['Hi', 'Hello', 'CARTO Rules', 'Shows dropdown'];
    const mapCard = shallowMount(MapCard, {
      propsData: { map },
      mocks: {
        $cartoModels,
        $t
      }
    });
    expect(mapCard).toMatchSnapshot();
  });

  describe('Methods', () => {
    it('should toggle selected state', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.isSelected).toBe(false);

      mapCard.vm.toggleSelection();

      expect(mapCard.emitted('toggleSelection')).toBeTruthy();
      expect(mapCard.emitted('toggleSelection')).toEqual([
        [{ map, isSelected: true }]
      ]);
    });

    describe('should toggle favorite state', () => {
      it('should call like when map is not favorited', () => {
        const map = visualization.visualizations[0];
        const mapCard = shallowMount(MapCard, {
          store,
          localVue,
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });
        expect(mapCard.vm.$props.map.liked).toBe(false);

        mapCard.vm.toggleFavorite();

        expect(actions['maps/like']).toHaveBeenCalled();
      });

      it('should call deleteLike when map is favorited', () => {
        const map = visualization.visualizations[1];
        const mapCard = shallowMount(MapCard, {
          store,
          localVue,
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });
        expect(mapCard.vm.$props.map.liked).toBe(true);

        mapCard.vm.toggleFavorite();

        expect(actions['maps/deleteLike']).toHaveBeenCalled();
      });
    });

    it('should open quick actions', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.areQuickActionsOpen).toBe(false);

      mapCard.vm.openQuickActions();

      expect(mapCard.vm.areQuickActionsOpen).toBe(true);
      expect(mapCard).toMatchSnapshot();
    });

    it('should close quick actions', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.areQuickActionsOpen).toBe(false);
      mapCard.vm.openQuickActions();
      expect(mapCard.vm.areQuickActionsOpen).toBe(true);

      mapCard.vm.closeQuickActions();

      expect(mapCard.vm.areQuickActionsOpen).toBe(false);
      expect(mapCard).toMatchSnapshot();
    });

    it('should toggle mouse hover', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.activeHover).toBe(true);

      mapCard.find('.card-select').trigger('mouseover');

      expect(mapCard.vm.activeHover).toBe(false);
      expect(mapCard).toMatchSnapshot();

      mapCard.find('.card-select').trigger('mouseleave');

      expect(mapCard.vm.activeHover).toBe(true);
      expect(mapCard).toMatchSnapshot();
    });

    it('should show thumbnail error', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.isThumbnailErrored).toBe(false);

      mapCard.vm.onThumbnailError();

      expect(mapCard.vm.isThumbnailErrored).toBe(true);
      expect(mapCard).toMatchSnapshot();
    });
  });

  describe('Events', () => {
    it('should receive and process correctly the event when dropdown is open', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.areQuickActionsOpen).toBe(false);

      mapCard.find(MapQuickActions).vm.$emit('open');

      expect(mapCard.vm.areQuickActionsOpen).toBe(true);
      expect(mapCard).toMatchSnapshot();
    });

    it('should receive and process correctly the event when dropdown is closed', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      expect(mapCard.vm.areQuickActionsOpen).toBe(false);

      mapCard.vm.openQuickActions();
      expect(mapCard.vm.areQuickActionsOpen).toBe(true);
      mapCard.find(MapQuickActions).vm.$emit('close');

      expect(mapCard.vm.areQuickActionsOpen).toBe(false);
      expect(mapCard).toMatchSnapshot();
    });
  });

  describe('Computed', () => {
    it('should return correct css class', () => {
      const map = { ...visualization.visualizations[0] };
      map.privacy = 'PRIVATE';
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });
      map.privacy = 'PUBLIC';
      const mapCard2 = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });

      const result1 = mapCard.vm.privacyIcon;
      const result2 = mapCard2.vm.privacyIcon;

      expect(result1).toBe('icon--private');
      expect(result2).toBe('icon--public');
    });

    it('should return correct map thumbnail', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });

      expect(mapCard.vm.mapThumbnailUrl).toBe('http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png');
    });

    describe('tagsLength', () => {
      it('should return 0 tags', () => {
        const map = visualization.visualizations[0];
        const mapCard = shallowMount(MapCard, {
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(mapCard.vm.tagsLength).toBe(0);
      });

      it('should return number of tags', () => {
        const map = { ...visualization.visualizations[0] };
        map.tags = ['one', 'two', 'three with a space', 'fourth with a comma'];
        const mapCard = shallowMount(MapCard, {
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(mapCard.vm.tagsLength).toBe(4);
      });
    });

    it('should return correct map url', () => {
      const map = visualization.visualizations[0];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });

      expect(mapCard.vm.vizUrl).toBe('http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200');
    });

    describe('showViews', () => {
      it('should return true if the map has public privacy', () => {
        const map = {...visualization.visualizations[0]};
        map.privacy = 'PUBLIC';
        const publicMap = shallowMount(MapCard, {
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(publicMap.vm.showViews).toBe(true);
      });

      it('should return true if the map has link privacy', () => {
        const map = {...visualization.visualizations[0]};
        map.privacy = 'LINK';
        const publicMap = shallowMount(MapCard, {
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(publicMap.vm.showViews).toBe(true);
      });

      it('should return true if the map has password privacy', () => {
        const map = {...visualization.visualizations[0]};
        map.privacy = 'PASSWORD';
        const publicMap = shallowMount(MapCard, {
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(publicMap.vm.showViews).toBe(true);
      });

      it('should return false if the map has a privacy different than public/password/link', () => {
        const map = {...visualization.visualizations[0]};
        map.privacy = 'PRIVATE';
        const publicMap = shallowMount(MapCard, {
          propsData: { map },
          mocks: {
            $cartoModels,
            $t
          }
        });

        expect(publicMap.vm.showViews).toBe(false);
      });
    });

    it('should return the total amount of views', () => {
      const map = {...visualization.visualizations[0]};
      map.stats = {'2018-10-17': 0, '2018-10-18': 3, '2018-10-19': 0, '2018-10-20': 0, '2018-10-21': 0, '2018-10-22': 0, '2018-10-23': 0, '2018-10-24': 0, '2018-10-25': 0, '2018-10-26': 0, '2018-10-27': 0, '2018-10-28': 0, '2018-10-29': 0, '2018-10-30': 0, '2018-10-31': 0, '2018-11-01': 0, '2018-11-02': 20, '2018-11-03': 0, '2018-11-04': 0, '2018-11-05': 0, '2018-11-06': 0, '2018-11-07': 0, '2018-11-08': 0, '2018-11-09': 0, '2018-11-10': 0, '2018-11-11': 0, '2018-11-12': 100, '2018-11-13': 0, '2018-11-14': 0, '2018-11-15': 1};
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });

      expect(mapCard.vm.numberViews).toBe(124);
    });
    it('should return the correct amount of tag chars', () => {
      const map = {...visualization.visualizations[0]};
      map.tags = ['abc', 'bcd', '', 'with space'];
      const mapCard = shallowMount(MapCard, {
        propsData: { map },
        mocks: {
          $cartoModels,
          $t
        }
      });

      expect(mapCard.vm.tagsChars).toBe(22);
    });
  });

  describe('properties', () => {
    let map, mocks;
    beforeEach(() => {
      map = {...visualization.visualizations[0]};
      mocks = { $cartoModels, $t };
    });

    it('should select the map when the user clicks on the card and selectMode property is true', () => {
      const mapCardWrapper = shallowMount(MapCard, {
        propsData: { selectMode: true, map },
        mocks
      });

      mapCardWrapper.trigger('click');

      expect(mapCardWrapper.emitted().toggleSelection).toBeDefined();
    });

    it('should not the map when the user clicks on the card and selectMode property is true', () => {
      const mapCardWrapper = shallowMount(MapCard, {
        propsData: { selectMode: false, map },
        mocks
      });

      mapCardWrapper.trigger('click');

      expect(mapCardWrapper.emitted().toggleSelection).toBeUndefined();
    });
  });
});
