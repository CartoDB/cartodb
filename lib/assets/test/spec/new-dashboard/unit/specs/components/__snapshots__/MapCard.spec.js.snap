// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapCard.vue Events should receive and process correctly the event when dropdown is closed 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Events should receive and process correctly the event when dropdown is open 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--quick-actions-open card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should close quick actions 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should open quick actions 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--quick-actions-open card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should show thumbnail error 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--can-hover">
  <div class="card-media has-error">
    <!---->
    <div class="MapCard-error"></div>
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should toggle mouse hover 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--child-hover card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue Methods should toggle mouse hover 2`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue should render correct contents 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--selected card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue should render description and tag dropdown 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p title="Lorem ipsum dolor sit amet, consectetur adipiscing elit." class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <!---->
        <featuresdropdown-stub list="Hi,Hello,CARTO Rules,Shows dropdown" feature="tag" linkroute="tagSearch"><span class="feature-text text is-caption is-txtGrey">4 MapCard.tags</span></featuresdropdown-stub>
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue should render description and tags 1`] = `
<a href="http://example.com/viz/e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--can-hover">
  <div class="card-media"><img src="http://hello.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div> <span class="checkbox card-select"><input type="checkBox" class="checkbox-input"> <span class="checkbox-decoration"><img svg-inline="" src="../assets/icons/common/checkbox.svg"></span></span>
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p title="Lorem ipsum dolor sit amet, consectetur adipiscing elit." class="card-description text is-caption">Lorem ipsum dolor sit amet, consectetur adipiscing elit.</p>
    </div>
    <ul class="card-metadata">
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--public"></span>
        <p>MapCard.shared.PUBLIC <span>| MapCard.views</span></p>
      </li>
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li>
            <router-link-stub to="[object Object]">Hi</router-link-stub><span>, </span>
          </li>
          <li>
            <router-link-stub to="[object Object]">Hello</router-link-stub><span>, </span>
          </li>
          <li>
            <router-link-stub to="[object Object]">CARTO Rules</router-link-stub>
            <!---->
          </li>
          <!---->
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;

exports[`MapCard.vue should show the map is shared 1`] = `
<a href="http://cdb.localhost.lan:3000/viz/hello.e97e0001-f1c2-425e-8c9b-0fb28da59200" class="card map-card card--can-hover">
  <div class="card-media"><img src="http://cdb.example.com/api/v1/map/static/named/tpl_e97e0001_f1c2_425e_8c9b_0fb28da59200/600/280.png">
    <!---->
  </div>
  <!---->
  <div class="card-actions">
    <mapquickactions-stub map="[object Object]"></mapquickactions-stub>
  </div>
  <div class="card-text">
    <div class="card-header">
      <h2 title="First Map" class="card-title title is-caption">
        First Map&nbsp;
        <span class="card-favorite"><img svg-inline="" src="../assets/icons/common/favorite.svg"></span></h2>
      <p class="card-description text is-caption is-txtSoftGrey">MapCard.noDescription</p>
    </div>
    <ul class="card-metadata">
      <!---->
      <li class="card-metadataItem text is-caption"><span class="icon icon--privacy icon--sharedBy" style="background-image: url(/assets/unversioned/images/avatars/avatar_mountain_green.png);"></span>
        <p>MapCard.sharedBy</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/calendar.svg"></span>
        <p>MapCard.lastUpdate</p>
      </li>
      <li class="card-metadataItem text is-caption"><span class="icon"><img inline-svg="" src="../assets/icons/maps/tag.svg"></span>
        <ul class="card-tagList">
          <li><span>MapCard.noTags</span></li>
        </ul>
        <!---->
      </li>
    </ul>
  </div>
</a>
`;
